packages/wallet/src/services/DRepRegistrationTracker.ts

```diff
@@ -0,0 +1,31 @@
+import { Cardano } from '@cardano-sdk/core';
+import { Observable, map } from 'rxjs';
+import { TrackerSubject } from '@cardano-sdk/util-rxjs';
+
+export interface CreateDRepRegistrationTrackerProps {
+  historyTransactions$: Observable<Cardano.HydratedTx[]>;
+}
+
+export const createDRepRegistrationTracker = ({
+  historyTransactions$
+}: CreateDRepRegistrationTrackerProps): TrackerSubject<boolean> =>
+  new TrackerSubject(
+    historyTransactions$.pipe(
+      map((txs) => {
+        const reverseTxs = txs.reverse();
```

Comment:
If I remember correctly, `txs` emitted by `historyTransactions$` is _constant_ `Array`; since `Array.reverse()` reverses the array in place, this risks to _break_ the source array.
```suggestion
        const reverseTxs = [...txs].reverse();
```
I'm not 100% sure, I am right, but I'm neither 100% sure this line is correct.

Reply:
yeh agreed we should clone via spread

Reply:
done